### PR TITLE
Haozhe dev

### PR DESCRIPTION
**Pull request by zhuhaozhe**
OneDNN dose not support Double type.

### CODE REVIEW

File: torch_ipex/csrc/cpu/dbl/Common.cpp [hunk 1]
```diff
@@ -17,9 +17,14 @@ namespace dbl {
 namespace comm {
 
 dil::tensor dil_tensor_from_dense(const at::Tensor& tensor) {
-  AT_ASSERTM(
+  TORCH_CHECK(
     tensor.layout() == at::Layout::Strided,
     "dil_tensor_view_from_dense expects dense tensor input");
+  TORCH_CHECK(tensor.scalar_type() == at::ScalarType::Float ||
+             tensor.scalar_type() == at::ScalarType::BFloat16,
+             "dil_tensor_view_from_dense expects bfloat16 or float tensor input");
```
Comment (pinzhenx) [inline torch_ipex/csrc/cpu/dbl/Common.cpp:25]: error message was not correct

File: torch_ipex/csrc/cpu/dbl/Common.cpp [hunk 2]
```diff
@@ -17,9 +17,14 @@ namespace dbl {
 namespace comm {
 
 dil::tensor dil_tensor_from_dense(const at::Tensor& tensor) {
-  AT_ASSERTM(
+  TORCH_CHECK(
     tensor.layout() == at::Layout::Strided,
     "dil_tensor_view_from_dense expects dense tensor input");
+  TORCH_CHECK(tensor.scalar_type() == at::ScalarType::Float ||
+             tensor.scalar_type() == at::ScalarType::BFloat16,
```
Comment (pinzhenx) [inline torch_ipex/csrc/cpu/dbl/Common.cpp:24]: @EikanWang from pytorch side, the only datatype we accept should only be `Float`, right?

File: torch_ipex/csrc/cpu/dbl/Common.cpp (hunk 1 above)
Comment (zhuhaozhe) [reply torch_ipex/csrc/cpu/dbl/Common.cpp:25]: What it should be? "dil_tensor_from_dense expects ......."?

File: torch_ipex/csrc/cpu/dbl/Common.cpp [hunk 3]
```diff
@@ -17,9 +17,14 @@ namespace dbl {
...
+             "dil_tensor_view_from_dense expects bfloat16 or float tensor input");
+  TORCH_CHECK(tensor.dim() <= 5,
+             "Can't convert cpu tensor with the number of dimensions > 5");
```
Comment (pinzhenx) [inline torch_ipex/csrc/cpu/dbl/Common.cpp:27]: Could you please try to remove this assertion in transpose and verify if the `test_torch` passes?

https://github.com/intel/intel-extension-for-pytorch/blob/e91c403d19699f5832ce12c460634bfdbc556aec/torch_ipex/csrc/cpu/DevOPs.cpp#L1093

I think this PR should cover that case.

**Comment by pinzhenx** [comment]
For unsupported datatype, I thought it would be better to throw an error here:

https://github.com/intel/intel-extension-for-pytorch/blob/master/torch_ipex/csrc/utils.cpp#L84

File: torch_ipex/csrc/cpu/dbl/Common.cpp (hunk 2 above)
Comment (EikanWang) [reply torch_ipex/csrc/cpu/dbl/Common.cpp:24]: yes

File: torch_ipex/csrc/cpu/dbl/Common.cpp (hunk 1 above)
Comment (pinzhenx) [reply torch_ipex/csrc/cpu/dbl/Common.cpp:25]: Yes. And the input scalar type should only be `at::ScalarType::Float`

**Comment by zhuhaozhe** [comment]
HIi, @pinzhenx @EikanWang please review

File: torch_ipex/csrc/cpu/dbl/Common.cpp [hunk 4]
```diff
@@ -17,9 +17,13 @@ namespace dbl {
 namespace comm {
 
 dil::tensor dil_tensor_from_dense(const at::Tensor& tensor) {
-  AT_ASSERTM(
+  TORCH_CHECK(
     tensor.layout() == at::Layout::Strided,
-    "dil_tensor_view_from_dense expects dense tensor input");
+    "dil_tensor_from_dense expects dense tensor input");
+  TORCH_CHECK(tensor.scalar_type() == at::ScalarType::Float
+             "dil_tensor_from_dense expects  tensor input");
```
Comment (pinzhenx) [inline torch_ipex/csrc/cpu/dbl/Common.cpp:24]: `dil_tensor_from_dense expects tensor of at::ScalarType::Float`

Comment (zhuhaozhe) [reply torch_ipex/csrc/cpu/dbl/Common.cpp]: Done

**Pull request merged**